src/__tests__/x509/asn1/obj.test.ts

```diff
@@ -0,0 +1,206 @@
+import { ASN1TypeError } from '../../../x509/asn1/error';
+import { ASN1Obj } from '../../../x509/asn1/obj';
+
+describe('ASN1Obj', () => {
+  describe('parseBuffer', () => {
+    describe('when parsing a primitive', () => {
+      const buffer = Buffer.from('02021010', 'hex');
+
+      it('parses a primitive', () => {
+        const obj = ASN1Obj.parseBuffer(buffer);
+
+        expect(obj).toBeInstanceOf(ASN1Obj);
+        expect(obj.tag.number).toBe(2);
+        expect(obj.value.toString('hex')).toBe('1010');
+        expect(obj.subs).toHaveLength(0);
+        expect(obj.raw).toEqual(buffer);
+      });
+    });
+
+    describe('when parsing a constructed object', () => {
+      const buffer = Buffer.from('30080202101002021111', 'hex');
```

Comment:
ooc where are these values coming from?

Reply:
Just some test data I invented to exercise all the various parsing paths. I added some comments which better explain what the data represents.